static/app/views/onboarding/components/heartbeatFooter/index.tsx

```diff
@@ -135,6 +130,20 @@ export function HeartbeatFooter({
     router.setRouteLeaveHook(route, onUnload);
   }, [serverConnected, firstErrorReceived, route, router, organization.slug, location]);
 
+  useEffect(() => {
+    if (loading) {
+      return;
+    }
+
+    if (serverConnected) {
+      addSuccessMessage(t('DSN response received'));
+    }
+
+    if (firstErrorReceived) {
+      addSuccessMessage(t('First error received'));
+    }
```

Comment:
Will you not fire both messages the second time this fires?

If the first time only `serverConnected` is true, you'll get `DSN response received`. But then the next time when `firstErrorReceived` becomes true, and this effect fires again, you'll get two success messages, because now both `serverConnected` and `firstErrorReceived` are true.

You probably want these in two separate use effects.

Reply:
I thought the same but everything runs too quickly that this doesn't seem to be an issue. When this case happens, the message display is the last one replacing the first.

Reply:
I think I would still be prudent and separate them.

you could imagine a world where we allow multiple toasts, in which case we would end up showing both. As it is right now, others may stumble on this code with the same question I had.

Reply:
Code updated 😉